File: 2015/c/day-01.c
#include <stdio.h>
#include <stdbool.h>

int main(void)
{
    char ch;
    int pos = 0;
    int floor = 0;
    bool found = false;

    FILE *fp = fopen("../input/day-01.txt", "r");

    if (fp == NULL) {
        printf("Unable to open file.");
        return 1;
    }

    while ((ch = fgetc(fp)) != EOF) {
        if (!found) pos++;
        if (ch == '(') floor++;
        if (ch == ')') floor--;
        if (floor == -1) found = true;
    }

    fclose(fp);

    printf("Part 1:  %d\nPart 2: %d\n", floor, pos);
    return 0;
}

File: 2015/perl/day-01.pl
use 5.42.0;
open my $fh, '<', '../input/day-01.txt';
my ( $floor, $pos );
while ( defined( my $char = getc($fh) ) ) {
    $char eq '(' and $floor++;
    $char eq ')' and $floor--;
    $pos = tell $fh if $floor == -1 and !defined $pos;
}
say "Part 1:  $floor\nPart 2: $pos";
